File: frontend/src/api/studentService.js
import axiosInstance from "./axiosInstance";

// 🔹 Get All Students
export const getStudents = async () => {
  const res = await axiosInstance.get("/api/students");
  return res.data?.data ?? res.data;
};

// 🔹 Get Students By Batch
export const getStudentsByBatch = async (batchId) => {
  const res = await axiosInstance.get(
    `/api/students/batch/${batchId}`
  );
  return res.data?.data ?? res.data;
};

// 🔹 Search Students (by name / email / enrollment)
export const searchStudents = async (q) => {
  const res = await axiosInstance.get(
    "/api/students/search",
    {
      params: { q },   // better than string concat
    }
  );
  return res.data?.data ?? res.data;
};

// 🔹 Dashboard Summary
export const getSummary = async () => {
  const res = await axiosInstance.get(
    "/api/students/summary"
  );
  return res.data?.data ?? res.data;
};

// 🔹 Get Single Student Detail
export const getStudentDetail = async (id) => {
  const res = await axiosInstance.get(
    `/api/students/${id}`
  );
  return res.data?.data ?? res.data;
};

File: backend/src/main/java/com/navkalpana/services/StudentService.java
package com.navkalpana.services;

import com.navkalpana.dto.respose.StudentDetailResponse;
import com.navkalpana.dto.respose.StudentListResponse;
import com.navkalpana.dto.respose.StudentSummaryResponse;
import com.navkalpana.entity.Student;

import java.util.List;

public interface StudentService {

    // 🔹 Add single student (admin use)
    Student saveRaw(Student s);

    // 🔹 Bulk insert
    List<Student> saveAllStudents(List<Student> students);

    // 🔹 Get all students
    List<StudentListResponse> getAll();

    // 🔹 Search by name / email / enrollment / batch
    List<StudentListResponse> search(String q);

    // 🔹 Filter by Batch
    List<StudentListResponse> filterByBatch(Long batchId);

    // 🔹 Dashboard Summary
    StudentSummaryResponse summary();

    // 🔹 Student Detail View
    StudentDetailResponse detail(Long id);
}